sdk/communication/azure-communication-callingserver/tests.yml

```diff
@@ -1,8 +1,32 @@
 trigger: none
 
 stages:
-  - template: /sdk/communication/communication-tests-template.yml
+  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
     parameters:
-      PackageName: azure-communication-callingserver
-      SafeName: azurecommunicationcallingserver
+      CloudConfig:
+        Public:
+          SubscriptionConfigurations:
+            - $(sub-config-azure-cloud-test-resources)
+            - $(sub-config-communication-services-cloud-test-resources-common)
+            - $(sub-config-communication-services-cloud-test-resources-java)
+      Clouds: Public
+      Artifacts:
+        - name: azure-communication-callingserver
+          groupId: com.azure
+          safeName: azurecommunicationcallingserver
+      ServiceDirectory: communication
+      PreSteps:
+        - bash: echo "##vso[task.setvariable variable=DefaultTestOptions]-Djacoco.skip=true $(DefaultTestOptions)"
+          condition: not(startsWith(variables['System.StageName'], 'Public'))
+      PostSteps:
+        - task: PublishCodeCoverageResults@1
+          condition: and(eq(variables['Agent.OS'], 'Windows_NT'), eq(variables['JavaTestVersion'], '1.11'), startsWith(variables['System.StageName'], 'Public'))
+          inputs:
+            codeCoverageTool: JaCoCo
+            summaryFileLocation: sdk/communication/azure-communication-callingserver/target/site/jacoco/jacoco.xml
+            reportDirectory: sdk/communication/azure-communication-callingserver/target/site/jacoco/
+            failIfCoverageEmpty: true
+      EnvVars:
+        SKIP_LIVE_TEST: TRUE
+        CALLINGSERVER_GENERATE_RANDOM_GROUP_IDENTIFIER: TRUE
```

Comment:
Are you copying the whole template so you can add the extra env var and/or remove int from the cloud config? I think it would be better to add support for passing these both in as parameters, e.g.:

tests.yml
```
 - template: /sdk/communication/communication-tests-template.yml
     parameters:
       Clouds: Public
       EnvVars:
         CALLINGSERVER_GENERATE_RANDOM_GROUP_IDENTIFIER: TRUE
```

/sdk/communication/communication-tests-template.yml
```
parameters:
  PackageName: not-specified
  SafeName: notspecified
  Cloud: 'Public,Int'
  EnvVars: {}

stages:
  - template: /eng/pipelines/templates/stages/archetype-sdk-tests.yml
    parameters:
      Clouds: ${{ parameters.Clouds }}

... other stuff ...

      EnvVars:
        SKIP_LIVE_TEST: TRUE
        ${{ each var in parameters.EnVars }}:
          ${{ var.key }}: ${{ var.value }}

```

Reply:
Let me try that out

Reply:
Fixed.